core/commands/commands.go

```diff
@@ -192,3 +193,14 @@ func unwrapOutput(i interface{}) (interface{}, error) {
 
 	return <-ch, nil
 }
+
+// closeUnlessErr closes `c`, unless `req` encountered an error (e.g. a timeout
+// or a cancellation). `defer closeUnlessErr(ctx, c)` is useful as an
+// alternative for `defer close(c)` if you want to prevent the receiver from
+// seeing the closed channel when there was a context error (which might
+// accidentally let it assume that there was no context error).
+func closeUnlessErr(ctx context.Context, c chan interface{}) {
```

Comment:
Not sure about the name, it doesn't really make it immediately obvious what the method does.

Reply:
I agree but I can't think of a better name...

Reply:
I think `maybeCloseChannel` could work, we use the `maybe` phrase elsewhere in go-ipfs for similar things.

Reply:
IMO, that's worse. We usually use maybe to mean "do this if we need to". In this case, we're saying "do this unless we shouldn't".